extensions/desktop-common/desktop-init

```diff
@@ -0,0 +1,57 @@
+#!/bin/bash -e
+
+#################
+# Launcher init #
+#################
+
+# On Fedora $SNAP is under /var and there is some magic to map it to /snap.
+# # We need to handle that case and reset $SNAP
+SNAP="${SNAP/\/var\/lib\/snapd/}"
+
+SNAP_DESKTOP_COMPONENTS_NEED_UPDATE="true"
+
+# shellcheck source=/dev/null
+. "$SNAP_USER_DATA/.last_revision" 2>/dev/null || true
+if [ "$SNAP_DESKTOP_LAST_REVISION" = "$SNAP_REVISION" ]; then
+  SNAP_DESKTOP_COMPONENTS_NEED_UPDATE="false"
+fi
+
+# Set $REALHOME to the users real home directory
+REALHOME="$(getent passwd $UID | cut -d ':' -f 6)"
+
+# If the user has modified their user-dirs settings, force an update
+if [[ -f "$XDG_CONFIG_HOME/user-dirs.dirs.md5sum" && -f "$XDG_CONFIG_HOME/user-dirs.locale.md5sum" ]]; then
+  if [[ "$(md5sum < "$REALHOME/.config/user-dirs.dirs")" != "$(cat "$XDG_CONFIG_HOME/user-dirs.dirs.md5sum")" ||
+        "$(md5sum < "$REALHOME/.config/user-dirs.locale")" != "$(cat "$XDG_CONFIG_HOME/user-dirs.locale.md5sum")" ]]; then
+    SNAP_DESKTOP_COMPONENTS_NEED_UPDATE="true"
+  fi
+fi
+
+if [ "$SNAP_ARCH" == "amd64" ]; then
+  ARCH="x86_64-linux-gnu"
+elif [ "$SNAP_ARCH" == "armhf" ]; then
+  ARCH="arm-linux-gnueabihf"
+elif [ "$SNAP_ARCH" == "arm64" ]; then
+  ARCH="aarch64-linux-gnu"
+elif [ "$SNAP_ARCH" == "ppc64el" ]; then
+  ARCH="powerpc64le-linux-gnu"
+else
+  ARCH="$SNAP_ARCH-linux-gnu"
+fi
+
+SNAP_LAUNCHER_ARCH_TRIPLET="$ARCH"
+
+if [ -n "$SNAP_DESKTOP_RUNTIME" ]; then
+  WITH_SNAP_DESKTOP_RUNTIME="yes"
+fi
+
+if [ -f "$SNAP/lib/bindtextdomain.so" ]; then
+  export LD_PRELOAD="$LD_PRELOAD:$SNAP/lib/bindtextdomain.so"
+fi
+
+export REALHOME
+export WITH_SNAP_DESKTOP_RUNTIME
+export SNAP_DESKTOP_COMPONENTS_NEED_UPDATE
+export SNAP_LAUNCHER_ARCH_TRIPLET
```

Comment:
It seems worth picking a single prefix, and sticking with it, so that all content from the desktop extension is under a unique, clear, and hopefully safe namespace.

In that sense, most things are using $SNAP_DESKTOP_* here, which follows that and is a nice and readable pattern. $SNAP_LAUNCHER_* breaks that, though. Can we use $SNAP_DESKTOP_* here too?

$SNAP_* alone risks getting into interaction trouble between internal snapd logic.

Reply:
I've renamed `$SNAP_LAUNCHER_ARCH_TRIPLET` to `$SNAP_DESKTOP_ARCH_TRIPLET` and made relevant updates to the other scripts to suit via 364960e1